Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,385 @@
+<?php
+
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPRuntimeException;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\PhpAmqpLibQueueManager;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueManager;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+use Smartbox\Integration\FrameworkBundle\Service;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+/**
+ * Class PhpAmqpLibDriver
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers
+ */
+class PhpAmqpLibDriver extends Service implements PurgeableQueueDriverInterface
+{
+    use UsesSerializer;
+    use UsesExceptionHandlerTrait;
+    use ContainerAwareTrait;
+    
+    /**
+     * Maximum amount of time (in seconds) to wait for a message.
+     */
+    const WAIT_TIMEOUT = 10;
+
+    /**
+     * Maximum amount of time (in seconds) to set as an interval while consuming messages
+     */
+    const WAIT_PERIOD = 0.2;
+
+    /**
+     * @var \AMQPQueue
+     */
+    private $queue;
+
+    /**
+     * @var int Time in ms
+     */
+    private $dequeueingTimeMs = 0;
+
+    /**
+     * @var string
+     */
+    private $format;
+
+    /**
+     * @var PhpAmqpLibQueueManager
+     */
+    private $manager;
+
+    /**
+     * @var string Host form Message Broker
+     */
+    private $host;
+
+    /**
+     * @var string Username to connect to Message broker
+     */
+    private $username;
+
+    /**
+     * @var string $password
+     */
+    private $password;
+
+    /**
+     * @var string $port
+     */
+    private $port;
+
+    /**
+     * @var string $vhost
+     */
+    private $vhost;
+
+    /**
+     * @var array
+     */
+    private $connectionId = [];
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * PhpAmqpLibDriver constructor.
+     * @param QueueManager $manager
+     */
+    public function __construct(PhpAmqpLibQueueManager $manager)
+    {
+        parent::__construct();
+        $this->manager = $manager;
+        $this->format = self::FORMAT_JSON;
```

Comment:
Here it is assumed that the format this driver will manage the messages is json, but it could be xml or other format. This is defined in the ```configure``` method and then added on the ```DependencyInjection/SmartboxIntegrationFrameworkExtension.php``` based on what the user has defined in the driver configuration

Reply:
Yes. I altered this part to take the parameter based on configuration.